samza-core/src/main/java/org/apache/samza/clustermanager/JobCoordinatorLaunchUtil.java

```diff
@@ -59,6 +59,8 @@ public static void run(SamzaApplication app, Config config) {
     }
 
     Config fullConfig = jobConfigs.get(0);
+    // Create coordinator stream if does not exist before fetching launch config from it.
+    CoordinatorStreamUtil.createCoordinatorStream(fullConfig);
```

Comment:
From what i understand,  CoordinatorStreamUtil.createCoordinateStream() is idempotent right.. so calling this twice should not have any side effects. Asking since you are not removing the creation from later code paths but adding it here.. so want to be sure.

Reply:
Yes, we are actually running createCoordinateStream() every time in writeConfigToCoordinatorStream().

Reply:
Sounds inefficient and Kafka'ish. It will be good to re-evaluate this with the metadata store work on whether metadata store abstraction needs to have idempotent resource creation or expose a method to identify the existence of a resource for callers. Can you create a follow up JIRA?

Reply:
Agree, the existing CoordinatorStreamUtil is not following the MetadataStore interface, making it explicit to Kafka. Created SAMZA-2539 as a follow up ticket to consolidate the logic.